packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -13,30 +13,40 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { SpanAttributeValue, SpanAttributes } from '@opentelemetry/api';
 
-export function sanitizeAttributes(attributes: unknown): SpanAttributes {
-  const out: SpanAttributes = {};
+import { diag, AttributeValue, Attributes } from '@opentelemetry/api';
 
-  if (attributes == null || typeof attributes !== 'object') {
+export function sanitizeAttributes(attributes: unknown): Attributes {
+  const out: Attributes = {};
+
+  if (typeof attributes !== 'object' || attributes == null) {
     return out;
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  for (const [k, v] of Object.entries(attributes!)) {
-    if (isAttributeValue(v)) {
-      if (Array.isArray(v)) {
-        out[k] = v.slice();
-      } else {
-        out[k] = v;
-      }
+  for (const [k, v] of Object.entries(attributes)) {
+    if (!isAttributeKey(k)) {
+      diag.warn(`Invalid attribute key: ${k}`);
+      continue;
+    }
+    if (!isAttributeValue(v)) {
+      diag.warn(`Invalid attribute value set for key: ${k}`);
+      continue;
+    }
+    if (Array.isArray(v)) {
+      out[k] = v.slice();
+    } else {
+      out[k] = v;
     }
   }
 
   return out;
 }
 
-export function isAttributeValue(val: unknown): val is SpanAttributeValue {
+export function isAttributeKey(key: unknown): key is string {
+  return typeof key === 'string' && key.length > 0;
```

Comment:
Minor perf and minification improvements by reordering
```return key && typeof === 'string';```

Reply:
I didn't find there is any big perf difference between the two flavors: https://jsbench.me/30l1ntj53a/1

TBO, I think checking the length is more explicit and easier to read.